nimble/controller/src/ble_ll_sync.c

```diff
@@ -1117,6 +1257,14 @@ ble_ll_sync_rx_pkt_in(struct os_mbuf *rxpdu, struct ble_mbuf_hdr *hdr)
         /* send reports from this PDU */
         ble_ll_sync_send_per_adv_rpt(sm, rxpdu, hdr->rxinfo.rssi, tx_power,
                                      datalen, aux, aux_scheduled);
+
+#if MYNEWT_VAL(BLE_LL_CFG_FEAT_LL_BIGINFO_REPORTS)
+        /* send biginfo only if it is present and at least one per_adv_rpt event was sent */
+        if ((sm->flags & BLE_LE_SYNC_SM_FLAG_BIGINFO) &&
+            !(sm->flags & BLE_LL_SYNC_SM_FLAG_HCI_NO_DATA_SENT)) {
+            ble_ll_sync_send_biginfo_adv_rpt(sm);
+        }
```

Comment:
you just send biginfo report if it's present in current PDU, no need to use additional flags